File: 剑指Offer/数值的整数次方/min.java
public class Solution {
    public double Power(double base, int exponent) {
        double res = 1.0;
        boolean isna = false;
        if(exponent<0){
            isna = true;
            exponent = Math.abs(exponent);
        }
        while(exponent>0){
            res *= base;
            exponent--;
        }
        return isna == true?1/res:res;
  }
}

File: 剑指Offer/数值的整数次方/min.py
# -*- coding:utf-8 -*-
class Solution:
    def Power(self, base, exponent):
        # write code here
        curr = exponent if exponent >=0 else abs(exponent)
        res = 1.0
        while curr>0:
            res *= base
            curr -= 1
        return res if exponent >=0 else 1/res